packages/compass/src/main/auto-update-manager.ts

```diff
@@ -1,12 +1,233 @@
+
+      const answer = await dialog.showMessageBox({
+        icon: COMPASS_ICON,
+        title: 'New version available',
+        message: 'A new version of Compass is available to install',
+        detail: `Compass ${updateInfo.to} is available – you are currently using ${updateInfo.from}. Would you like to download and install it now?`,
+        buttons: ['Install', 'Ask me later'],
+        cancelId: 1,
+      });
+
+      if (answer.response === 0) {
+        updateManager.setState(
+          AutoUpdateManagerState.DownloadingUpdate,
+          updateInfo
+        );
+      } else {
+        updateManager.setState(
+          AutoUpdateManagerState.UpdateDismissed,
+          updateInfo
+        );
+      }
+    },
+    [AutoUpdateManagerState.NoUpdateAvailable]: async function (updateManager) {
+      log.info(
+        mongoLogId(1001000126),
+        'AutoUpdateManager',
+        'Update not available'
+      );
+      await wait(updateManager.autoUpdateOptions.updateCheckInterval);
+      if (this.aborted) {
+        return;
+      }
+      updateManager.setState(AutoUpdateManagerState.CheckingForUpdates);
+    },
+    [AutoUpdateManagerState.Disabled]: disableAutoUpdates,
+  },
+  [AutoUpdateManagerState.UpdateAvailable]: {
+    [AutoUpdateManagerState.DownloadingUpdate]: function (
+      updateManager,
+      updateInfo
+    ) {
+      track('Autoupdate Accepted', { update_version: updateInfo.to });
+
+      autoUpdater.once('error', (error) => {
+        updateManager.setState(AutoUpdateManagerState.DownloadingError, error);
+      });
+
+      autoUpdater.once('update-downloaded', () => {
+        updateManager.setState(
+          AutoUpdateManagerState.ReadyToUpdate,
+          updateInfo
+        );
+      });
+      // checkForUpdate downloads and installs the update when available, there
+      // is also no way to interrupt this process so once it starts, disabling
+      // updates in the options will not do anything
+      autoUpdater.setFeedURL(updateManager.getFeedURLOptions());
+      autoUpdater.checkForUpdates();
+    },
+    [AutoUpdateManagerState.UpdateDismissed]: (_updateManager, updateInfo) => {
+      track('Autoupdate Dismissed', { update_version: updateInfo.to });
+    },
+  },
+  [AutoUpdateManagerState.DownloadingUpdate]: {
+    [AutoUpdateManagerState.ReadyToUpdate]: async (
+      updateManager,
+      updateInfo: { from: string; to: string }
+    ) => {
+      log.info(
+        mongoLogId(1001000128),
+        'AutoUpdateManager',
+        'Update downloaded',
+        {
+          releaseVersion: updateInfo.to,
+        }
+      );
+      const answer = await dialog.showMessageBox({
+        icon: COMPASS_ICON,
+        title: 'Restart to finish the update',
+        message: `Restart Compass to finish installing ${updateInfo.to}`,
+        detail:
+          'Closing this window without restarting may cause things to not work as intended.',
```

Comment:
Maybe we could replace "things"?

```suggestion
          'Closing this window without restarting may cause some of the features to not work as intended.',
```